test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetNewTemplatesTests.cs

```diff
@@ -0,0 +1,708 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using FluentAssertions;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace Dotnet.Integration.Test
+{
+    [Collection("Dotnet Integration Tests")]
+    public class DotnetNewTemplatesTests
+    {
+        private readonly MsbuildIntegrationTestFixture _fixture;
+
+        public DotnetNewTemplatesTests(MsbuildIntegrationTestFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [Theory]
+        [InlineData("", "", "", false)]
+        [InlineData("ProjectA", "Myoutput", "VB", false)]
+        [InlineData("", "", "C#", true)]
+        [InlineData("ProjectA", "Myoutput", "F#", true)]
+        public void Dotnet_New_ConsoleApp_Success(string projectName, string output, string lang, bool norestore)
```

Comment:
Do different languages have any meaninful difference to NuGet? I know that the SDK might bring in different props/targets, but ultimately it's still running NuGet's restore target, so only PackageReferences (and other item types) matter.

I feel like this is just running the same scenario multiple times, but without any real possibility of catching bugs than if only one ran.

Reply:
Ok. Addressed.